### ISSUE TITLE
Cannot find PowerShell provider

### ISSUE THREAD

**Issue by McAtkin** [voxpupuli/puppet-windows_firewall#112]
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: Enterprise 2019.3.0 Agent 6.0.7 and 6.10.1
- Ruby: Per agent versions
- Distribution:
- Module version: 3.0.0

## How to reproduce (e.g Puppet code you use)

```puppet
windowsfirewall { 'domain':
    ensure => 'present',
}
```

## What are you seeing

on puppet run

```
Error: Could not prefetch windowsfirewall provider 'powershell': undefined local variable or method `resource' for Puppet::Type::Windowsfirewall::ProviderPowershell:Class
Did you mean?  resources
Error: Failed to apply catalog: undefined local variable or method `resource' for Puppet::Type::Windowsfirewall::ProviderPowershell:Class
Did you mean?  resources
```

## What behaviour did you expect instead

Firewall domain profile forced to enabled

## Output log

```
Debug: Executing: 'C:\windows\system32\WindowsPowershell\v1.0\powershell.exe Get-NetFirewallProfile -profile "domain"'
Debug: Windowsfirewall found this hash of properties on the system: {:ensure=>:present, :default_inbound_action=>"NotConfigured", :default_outbound_action=>"NotConfigured", :allow_inbound_rules=>"NotConfigured", :allow_local_firewall_rules=>"NotConfigured", :allow_local_ipsec_rules=>"NotConfigured", :allow_user_apps=>"NotConfigured", :allow_user_ports=>"NotConfigured", :allow_unicast_response_to_multicast=>"NotConfigured", :notify_on_listen=>"False", :enable_stealth_mode_for_ipsec=>"NotConfigured", :log_file_name=>"%systemroot%\\system32\\LogFiles\\Firewall\\pfirewall.log", :log_max_size_kilobytes=>"4096", :log_allowed=>"False", :log_blocked=>"False", :log_ignored=>"NotConfigured", :disabled_interface_aliases=>"{NotConfigured}", :name=>"domain", :provider=>:powershell}
Debug: Executing: 'C:\windows\system32\WindowsPowershell\v1.0\powershell.exe Get-NetFirewallProfile -profile "private"'
Debug: Windowsfirewall found this hash of properties on the system: {:ensure=>:present, :default_inbound_action=>"NotConfigured", :default_outbound_action=>"NotConfigured", :allow_inbound_rules=>"NotConfigured", :allow_local_firewall_rules=>"NotConfigured", :allow_local_ipsec_rules=>"NotConfigured", :allow_user_apps=>"NotConfigured", :allow_user_ports=>"NotConfigured", :allow_unicast_response_to_multicast=>"NotConfigured", :notify_on_listen=>"False", :enable_stealth_mode_for_ipsec=>"NotConfigured", :log_file_name=>"%systemroot%\\system32\\LogFiles\\Firewall\\pfirewall.log", :log_max_size_kilobytes=>"4096", :log_allowed=>"False", :log_blocked=>"False", :log_ignored=>"NotConfigured", :disabled_interface_aliases=>"{NotConfigured}", :name=>"private", :provider=>:powershell}
Debug: Executing: 'C:\windows\system32\WindowsPowershell\v1.0\powershell.exe Get-NetFirewallProfile -profile "public"'
Debug: Windowsfirewall found this hash of properties on the system: {:ensure=>:present, :default_inbound_action=>"NotConfigured", :default_outbound_action=>"NotConfigured", :allow_inbound_rules=>"NotConfigured", :allow_local_firewall_rules=>"NotConfigured", :allow_local_ipsec_rules=>"NotConfigured", :allow_user_apps=>"NotConfigured", :allow_user_ports=>"NotConfigured", :allow_unicast_response_to_multicast=>"NotConfigured", :notify_on_listen=>"False", :enable_stealth_mode_for_ipsec=>"NotConfigured", :log_file_name=>"%systemroot%\\system32\\LogFiles\\Firewall\\pfirewall.log", :log_max_size_kilobytes=>"4096", :log_allowed=>"False", :log_blocked=>"False", :log_ignored=>"NotConfigured", :disabled_interface_aliases=>"{NotConfigured}", :name=>"public", :provider=>:powershell}
Error: Could not prefetch windowsfirewall provider 'powershell': undefined local variable or method `resource' for Puppet::Type::Windowsfirewall::ProviderPowershell:Class
Did you mean?  resources
Debug: Storing state
Debug: Pruned old state cache entries in 0.00 seconds
Debug: Stored state in 0.09 seconds
Error: Failed to apply catalog: undefined local variable or method `resource' for Puppet::Type::Windowsfirewall::ProviderPowershell:Class
Did you mean?  resources
```

**Comment by Nwolford51** [comment]
Having this same issue:  
`Error: Could not prefetch windowsfirewall provider 'powershell': undefined local variable or method `resource' for Puppet::Type::Windowsfirewall::ProviderPowershell:Class
Did you mean?  resources
C:/ProgramData/PuppetLabs/puppet/cache/lib/puppet/provider/windowsfirewall/powershell.rb:79:in `block in prefetch'
C:/ProgramData/PuppetLabs/puppet/cache/lib/puppet/provider/windowsfirewall/powershell.rb:78:in `each'
C:/ProgramData/PuppetLabs/puppet/cache/lib/puppet/provider/windowsfirewall/powershell.rb:78:in `prefetch'`

**Comment by msiroskey** [comment]
I’ve encountered this issue and after some investigation it appears to be a bug in puppet forge 3.0.0 release of this module.  This bug is patched in the master branch on github (I wish I noticed this after doing a few hours of debugging).  Basically, the resource variable should reference the matching instance in the resources hash but in the bad code it tries to evaluate that resource (never set) matches resources[prov.name].  Guessing it was an error in converting what is normally a multiline if into a single line if statement.

**Comment by kenyon** [comment]
Fixed in #104.
